File: Capstone_Project/Frontend/flightbookingapp/src/app/interface/booking.ts
import { PaymentStatus } from "../../paymentstatus";
export interface IBooking {
  id: number;             
  flightId: number;
  passengerName: string;
  email: string;
  amount: number;
  cardNumber?: string;
  bookingTime?: string;
  status?: PaymentStatus; 
}

File: Capstone_Project/bookingms/src/main/java/com/wipro/bookingms/entity/Booking.java
package com.wipro.bookingms.entity;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;


@Entity
@Table(name = "bookings")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Booking {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private Long flightId;
    private String passengerName;
    private String email;
    private Double amount;
    private String cardNumber;

    @Enumerated(EnumType.STRING)
    private PaymentStatus status;

    private LocalDateTime bookingTime;
}
